### ISSUE TITLE
DDEV on VMware Ubuntu 18 VM

### ISSUE THREAD

**Issue by sri-vinay** [drud/ddev#1826]
Hi, ddev will not start on VMware Ubuntu 18 VM and fails with the following error messages.

Failed to start drupaltraining: ddev-router failed to become ready: logOutput=nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
nginx config: OK  ddev-router healthcheck endpoint not responding
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 127.0.0.1 port 80: Connection refused
, err=container /ddev-router unhealthy: nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
nginx config: OK  ddev-router healthcheck endpoint not responding
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 127.0.0.1 port 80: Connection refused 

I was able to get it working on a physical Ubuntu 18 box.

Thanks for any help

**Comment by rfay** [comment]
It seems that your ddev-router is broken. Please take these steps:

1. `ddev poweroff`
2. Make sure all docker containers are destroyed with `docker rm -f $(docker ps -aq)`
3. Try a sample dummy project to see if you can start it up:

```
mkdir -p ~/tmp/junk
cd ~/tmp/junk
ddev config --php-version=7.2
ddev start
```

Does that have the same problem? If it comes up without any problem, please inspect your config.yaml from the misbehaving project; if you don't see what's wrong there, please share it here. 

`docker logs ddev-router` may also shed some light on the problem.

**Comment by sri-vinay** [comment]
Hi Randy, thanks for the quick response. I tried the suggestions and get the same problem. docker logs output this:

Using the local CA at "/root/.local/share/mkcert" ✨
The local CA is now installed in the system trust store! ⚡️

Using the local CA at "/root/.local/share/mkcert" ✨
ERROR: can't create new certificates because the CA key (rootCA-key.pem) is missing

Thanks again for the help!

Sri

**Comment by rfay** [comment]
Hmm, I assume that's the output of `docker logs ddev-router` ? (Try putting in ````` to make clearer)

Please give the full output of the startup of your junk project after you do the `ddev poweroff`, thanks.

**Comment by sri-vinay** [comment]
Yes, that was the output of docker logs ddev-router.

After ddev poweroff and ddev start gave the following output. test project is the one you suggested.

******
ddev start
Starting test...
Building ddev-ssh-agent
Creating ddev-ssh-agent ... done

ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys.
Pushing mkcert rootca certs to ddev-global-cache
failed to copy root CA into docker volume: container run failed with exit code 1, output=''
Pushed mkcert rootca certs to ddev-global-cache
Running   Command=ip address show dev docker0
Building db
Building web
Creating ddev-test-db ... done
Creating ddev-test-web ... done
Creating ddev-test-dba ... done

Creating ddev-router ... done

Failed to start test: ddev-router failed to become ready: logOutput=nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
nginx config: OK    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 127.0.0.1 port 80: Connection refused
ddev-router healthcheck endpoint not responding
, err=container /ddev-router unhealthy: nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
nginx config: OK    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 127.0.0.1 port 80: Connection refused
ddev-router healthcheck endpoint not responding
******

Thanks.

**Comment by rfay** [comment]
OK, please try `docker inspect --format "{{json .State.Health }}" ddev-router` (soon during the startup, you don't have to wait for it to tail)

It's possible of course that your environment doesn't allow dropping privileges to allow connection to port 80. I've run in many virtualized Ubuntu environments without any trouble, but if your docker or system environment doesn't allow "setcap CAP_NET_BIND_SERVICE=+eip /usr/sbin/nginx" then that might do this. You'll need to find out why it's not allowing that privilege. 

https://github.com/drud/ddev/blob/cb0d34fa7c7ecb8a6c11c244023e13227eb28f1c/containers/ddev-webserver/Dockerfile#L101

**Comment by sri-vinay** [comment]
Here is the output:

******
 docker inspect --format "{{json .State.Health }}" ddev-router
{"Status":"unhealthy","FailingStreak":0,"Log":[{"Start":"2019-09-11T15:08:24.266113548-04:00","End":"2019-09-11T15:08:24.891155654-04:00","ExitCode":1,"Output":"nginx: the configuration file /etc/nginx/nginx.conf syntax is ok\nnginx config: OK  nginx: configuration file /etc/nginx/nginx.conf test is successful\n  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 127.0.0.1 port 80: Connection refused\nddev-router healthcheck endpoint not responding \n"},{"Start":"2019-09-11T15:08:25.897606396-04:00","End":"2019-09-11T15:08:26.450871173-04:00","ExitCode":1,"Output":"nginx config: OK  nginx: the configuration file /etc/nginx/nginx.conf syntax is ok\nnginx: configuration file /etc/nginx/nginx.conf test is successful\n  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 127.0.0.1 port 80: Connection refused\nddev-router healthcheck endpoint not responding \n"}]}
******

I am looking into the setcap.

Thanks.

**Comment by rfay** [comment]
My bet is that your VM has a restricted setcap environment. I do think I heard of this one time previously.

How did you install docker? With the standard apt docker-ce repository? It also may be the VMWare environment.

**Comment by sri-vinay** [comment]
Yes, I did install docker using the standard apt docker-ce repo. I am fairly sure it is probably the VMware ESXi environment as you pointed me to...thanks! Looking into it.

Thanks.
